safekeeper/src/timeline_manager.rs

```diff
@@ -579,18 +602,36 @@ impl Manager {
         }
     }
 
+    /// Reset partial backup state and remove its remote storage data. Since it
+    /// might concurrently uploading something, cancel the task first.
+    async fn backup_partial_reset(&mut self) -> anyhow::Result<Vec<String>> {
+        // Unevict timeline if it is evicted before erasing partial backup
+        // state.
+        self.unevict_timeline().await?;
+
+        if let Some((handle, cancel)) = &mut self.partial_backup_task {
```

Comment:
It's probably possible to use https://docs.rs/tokio/1.40.0/tokio/task/struct.JoinHandle.html#method.abort here without a need to introduce `CancellationToken`.